integration/inspektor-gadget/advise_networkpolicy_test.go

```diff
@@ -74,19 +74,22 @@ func TestAdviseNetworkpolicy(t *testing.T) {
 					},
 				}
 
-				expectedEntry.Container = ""
+				expectedEntry.K8s.ContainerName = ""
 
 				normalize := func(e *networkTypes.Event) {
-					e.Container = ""
 					e.Timestamp = 0
-					e.Node = ""
 					e.Pid = 0
 					e.Tid = 0
 					e.PodIP = ""
 					e.DstEndpoint.Addr = ""
 					e.PodHostIP = ""
 					e.NetNsID = 0
 					e.MountNsID = 0
+
+					e.K8s.Node = ""
+					e.K8s.ContainerName = ""
+					// TODO: Verify container runtime and container name
```

Comment:
Yep, I would like first to study and understand the different container enrichers (PodInformer, OCI-annotations, etc), create some tests for them independently and then request the runtime information here. It is because [ig-k8s doesn't directly use the Runtime enricher](https://github.com/inspektor-gadget/inspektor-gadget/blob/a39e7b00cc2d114259f3d9121aa4232ac7f6e407/pkg/gadgettracermanager/gadgettracermanager.go#L251-L320), so I want to be sure the runtime information can always be guaranteed, either by adding the Runtime enricher to ig-k8s or to ensuring the same information can be retrieved by other means in the k8s context.

I'm already working on this. I generalized the TODO for future PRs from " Improve how Runtime.Container is enriched for CRI-O." to "Ensure Runtime information is guaranteed in ig-k8s (`kubectl-gadget`)".

Reply:
> It is because [ig-k8s doesn't directly use the Runtime enricher](https://github.com/inspektor-gadget/inspektor-gadget/blob/a39e7b00cc2d114259f3d9121aa4232ac7f6e407/pkg/gadgettracermanager/gadgettracermanager.go#L251-L320)

Make sense. Yeah, I see it is [used](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/container-collection/k8s.go#L132) via Kubernetes client so it isn't clear if all the information will be available.